pandas/core/indexes/datetimes.py

```diff
@@ -576,10 +583,11 @@ def union(self, other):
         -------
         y : Index or DatetimeIndex
         """
-        self._assert_can_do_setop(other)
```

Comment:
you can just remove this no? (whole method)

Reply:
The reason for this is solely for the docstrings. Since all the logic is now held in _union this was a way to publicly expose the union docstrings of Index subtypes. I was meaning to find a better way to do that, is there something like this done elsewhere in the library?

Reply:
you don’t need to do this as the docstrings will be inherited

Reply:
The problem is that currently not all the docstrings for the subclasses union methods are the same.

    pd.Index.union.__doc__ != pd.DatetimeIndex.union.__doc__
    pd.Index.union.__doc__ != pd.TimedeltaIndex.union.__doc__
    pd.TimedeltaIndex.union.__doc__ != pd.DatetimeIndex.union.__doc__

Reply:
so we need to make all of the subclass signatures the same, the doc-strings should simply be inherited. the subclass ones are likely wrong / outdated.